odl/discr/partition.py

```diff
@@ -0,0 +1,561 @@
+                'cell size not defined for irregular partitions. Use '
+                'cell_sizes() instead.')
+
+        on_boundary = self.nodes_on_boundary()
+        num_cells = []
+        for n, on_bdry in zip(self.shape, on_boundary):
+            num_on_bdry = sum(on for on in on_bdry)
+            num_cells.append(n - num_on_bdry / 2)
+
+        return self.extent() / num_cells
+
+    @property
+    def cell_volume(self):
+        """Volume of the 'inner' cells, regardless of begin and end.
+
+        Only defined if ``self.grid`` is a `RegularGrid`.
+
+        Examples
+        --------
+        >>> grid = RegularGrid([0, 0], [1, 1], (3, 5))
+        >>> part = RectPartition(grid)
+        >>> part.cell_volume
+        0.125
+        """
+        return float(np.prod(self.cell_size))
+
+    def approx_equals(self, other, atol):
+        """Return `True` in case of approximate equality.
+
+        Returns
+        -------
+        approx_eq : `bool`
+            `True` if ``other`` is a `RectPartition` instance with
+            ``self.bbox == other.bbox`` up to ``atol`` and
+            ``self.grid == other.other`` up to ``atol``.
+        """
+        if other is self:
+            return True
+        elif not isinstance(other, RectPartition):
+            return False
+        else:
+            return (self.bbox.approx_equals(other.bbox, atol=atol) and
+                    self.grid.approx_equals(other.grid, atol=atol))
+
+    def __eq__(self, other):
+        """Return ``self == other``."""
+        return self.bbox == other.bbox and self.grid == other.grid
+
+    def insert(self, index, other):
+        """Return a copy with ``other`` inserted before ``index``.
+
+        Parameters
+        ----------
+        index : `int`
+            Index of the dimension before which ``other`` is to
+            be inserted. Must fulfill ``0 <= index <= ndim``.
+        other : `RectPartition`
+            Partition to be inserted
+
+        Returns
+        -------
+        newpart : `RectPartition`
+            Partition with the inserted other partition
+
+        Examples
+        --------
+        """
+        newgrid = self.grid.insert(index, other.grid)
+        newbbox = self.bbox.insert(index, other.bbox)
+        new_b = newbbox.begin
+        new_e = newbbox.end
+        return RectPartition(newgrid, begin=new_b, end=new_e)
+
+    # TODO: pretty-print
+    def __repr__(self):
+        """Return ``repr(self)``."""
+        inner_str = '\n {!r}'.format(self.grid)
+        sep = ',\n '
+        if self._custom_begin:
+            inner_str += sep + 'begin={}'.format(
+                array1d_repr(self.bbox.begin))
+            sep = ', '
+        if self._custom_end:
+            inner_str += sep + 'end={}'.format(
+                array1d_repr(self.bbox.end))
+
+        return '{}({})'.format(self.__class__.__name__, inner_str)
+
+
+def uniform_partition(intv_prod, num_nodes, nodes_on_bdry=True):
```

Comment:
First comment: sure

Second:
That's what I did at first, but as I wrote in the beginning, it's debatable. In `uniform_discr`, it's not the default (partly due to the computational penalty).